### ISSUE TITLE
`openssl rehash` returns exit code of 3 with no errors

### ISSUE THREAD

**Issue by zfjagann** [openssl/openssl#6199]
`openssl rehash` returns an exit code of 3 even though it prints no errors.

I ran it on certs I copied out of the [icaclient arch-linux package](https://aur.archlinux.org/packages/icaclient/) and it returns an exit code of 3 with a few warnings:

```
zeal@zeal ~/tmp $ openssl rehash certs/
rehash: skipping tls-ca-bundle.pem,it does not contain exactly one certificate or CRL
rehash: skipping cert.0.pem,it does not contain exactly one certificate or CRL
rehash: skipping duplicate certificate in cert.36.pem
rehash: skipping duplicate certificate in cert.59.pem
rehash: skipping duplicate certificate in cert.76.pem
zeal@zeal ~/tmp $ echo $?
3
```

For comparison, `c_rehash` prints the same warnings, but has an exit code of 0
```
zeal@zeal ~/tmp $ c_rehash certs/
Doing certs/
WARNING: cert.0.pem does not contain a certificate or CRL: skipping
WARNING: Skipping duplicate certificate cert.36.pem
WARNING: Skipping duplicate certificate cert.59.pem
WARNING: Skipping duplicate certificate cert.76.pem
WARNING: Skipping duplicate certificate tls-ca-bundle.pem
zeal@zeal ~/tmp $ echo $?
0
```

I wasn't sure what a return code of 3 was, so I poked at `rehash.c` to find out.

Looks like neither the [`skipping duplicate certificate`](https://github.com/openssl/openssl/blob/master/apps/rehash.c#L128-L132) nor the [`does not contain exactly one certificate`](https://github.com/openssl/openssl/blob/master/apps/rehash.c#L246-L251) messages are errors, just warnings (neither increment the error count), but the return code is still 3.

If I delete the files that were skipped, the exit code is now 0, indicating that one or more of them was causing an issue, even though the warnings didn't indicate this.

Seems like one of the warnings should actually be an error, or something else is causing issues with one of the files that isn't logged, I'm not sure which.

**Comment by richsalz** [comment]
Nice detective work!  The "skipping duplicate" is a warning and should not bump the error code.  "Does not contain exactly one certificate" should remain an error.

**Comment by richsalz** [comment]
duplicates are benign and we can ignore them.  Files not  containing what we want shouldn't be ignored.  There's probably work to do on both the script and the program, but maybe not in this release.

**Comment by levitte** [comment]
`c_rehash` allows a PEM file with a cert chain and will simply hash on the first certificate (the leaf).
`c_rehash` also allows a PEM file with a combined certificate / cert chain and a CRL, something that does make sense for a CA that chooses to publish both their CA cert and CRL for their EE certs that way.  In this case, a hash link for both the first cert and for the CRL are produced.

`openssl rehash` currently doesn't allow this for the moment, but should.

**Comment by richsalz** [comment]
This is behavior first in 1.1.0, so removing the 1.1.1 milestone.

**Comment by otavio** [comment]
This has been released on 1.1.1; could this to be closed?

**Comment by levitte** [comment]
Yup
